### ISSUE TITLE
Incorrect excerpt and title even if the meta tag exist

### ISSUE THREAD

**Issue by FarmaanElahi** [mozilla/readability#480]
As per the code ,priority is given to the meta tag and if that doesnot exist,library fallsback to extracted title and descrition.
```
if ("description" in values) {
      metadata.excerpt = values["description"];
    } else if ("og:description" in values) {
      // Use facebook open graph description.
      metadata.excerpt = values["og:description"];
    } else if ("twitter:description" in values) {
      // Use twitter cards description.
      metadata.excerpt = values["twitter:description"];
    }
```
```
// If we haven't found an excerpt in the article's metadata, use the article's
    // first paragraph as the excerpt. This is used for displaying a preview of
    // the article's content.
    if (!metadata.excerpt) {
      var paragraphs = articleContent.getElementsByTagName("p");
      if (paragraphs.length > 0) {
        metadata.excerpt = paragraphs[0].textContent.trim();
      }
    }
```

In cases,even if meta tag exist for `description` and `title` exist,it is not guarented that it is added to final result.

Way to reproduce:

- Open test page [links-in-tables](https://github.com/mozilla/readability/tree/master/test/test-pages/links-in-tables)
- In the source.html you will find meta tag for description
```
<meta content=" Posted by Andrew Hayden, Software Engineer on Google Play    Android users are downloading tens of billions of apps and games on Google Pla..." property="og:description" />
```
- But in the expected.metadata.json ,the description is 
```
Posted by Andrew Hayden, Software Engineer on Google Play
```

**Comment by FarmaanElahi** [comment]
Based on my observation: 
```
var namePattern = /^\s*((twitter)\s*:\s*)?(description|title|parsely-page|author)\s*$/gi;
var propertyPattern = /^\s*og\s*:\s*(description|title|image)\s*$/gi;
```
It is because of how JS  does the regex check  for groups,it is causing issue 
Look at this SO [post](https://stackoverflow.com/questions/209732/why-am-i-seeing-inconsistent-javascript-logic-behavior-looping-with-an-alert-v) for the regex implementation

**Comment by gijsk** [comment]
Thanks for the report! I put up a PR to fix this in #481.
